src/shared/utils/app/is-post-blocked.ts

```diff
@@ -6,12 +6,8 @@ export default function isPostBlocked(
   myUserInfo: MyUserInfo | undefined = UserService.Instance.myUserInfo,
 ): boolean {
   return (
-    (myUserInfo?.community_blocks
-      .map(c => c.community.id)
-      .includes(pv.community.id) ||
-      myUserInfo?.person_blocks
-        .map(p => p.target.id)
-        .includes(pv.creator.id)) ??
+    (myUserInfo?.community_blocks.map(c => c.id).includes(pv.community.id) ||
+      myUserInfo?.person_blocks.map(p => p.id).includes(pv.creator.id)) ??
```

Comment:
Nitpick: this can be rewritten as:

```suggestion
    (myUserInfo?.community_blocks.some(c => c.id === pv.community.id) ||
      myUserInfo?.person_blocks.some(p => p.id === pv.creator.id)) ??
```

Sadly, array methods in JavaScript aren't lazy like iterator methods in Rust are; this code iterates over all of the blocks to create a new array of just the IDs, then iterates through the new ID array to see if it includes the array you're looking for. Using `some` makes it so a new array isn't created and prevents you from needing to iterate over the blocks more than once. I'm not sure if the performance impact of this is noticeable barring extreme cases, but it's worth keeping in mind.

Reply:
Nice, thx.